File: Java/MonolithicApp/src/main/java/com/example/route/dto/HeatmapResponse.java
package com.example.route.dto;

import java.util.ArrayList;
import java.util.List;

/**
 * Response payload with aggregated edge usage counts.
 */
public class HeatmapResponse {
    private int totalRequests;
    private List<HeatmapEdgeUsage> edges = new ArrayList<>();
    private List<HeatmapRequest.RequestPair> missing = new ArrayList<>();

    public HeatmapResponse() {}

    public int getTotalRequests() { return totalRequests; }
    public void setTotalRequests(int totalRequests) { this.totalRequests = totalRequests; }

    public List<HeatmapEdgeUsage> getEdges() { return edges; }
    public void setEdges(List<HeatmapEdgeUsage> edges) { this.edges = edges; }

    public List<HeatmapRequest.RequestPair> getMissing() { return missing; }
    public void setMissing(List<HeatmapRequest.RequestPair> missing) { this.missing = missing; }
}

File: NET/MonolithicApp/MonolithicApp/DTOs/HeatmapResponse.cs
using System.Collections.Generic;

namespace MonolithicApp.DTOs
{
    public class HeatmapResponse
    {
        public int TotalRequests { get; set; }
        public List<HeatmapEdgeUsage> Edges { get; set; }
        public List<RequestPair> Missing { get; set; }
    }

    public class HeatmapEdgeUsage
    {
        public string From { get; set; }
        public string To { get; set; }
        public long Count { get; set; }

        public HeatmapEdgeUsage(string from, string to, long count)
        {
            From = from;
            To = to;
            Count = count;
        }
    }
}
